tools/single_use/SaveUserIDToInstrumentData.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * This script is intended for a one-time use only to restore the value of the
+ * `UserID` column of instrument tables and the `UserID` key of the instrument
+ * JSON `Data` in the flag table.
+ * 
+ * This tool queries data from the `history` table and meaningfully imports
+ * data from its `userID` column back into the instrument and `flag` tables.
+ *
+ * Example use:
+ * $ php SaveUserIDToInstrumentData.php [confirm]
+
+ * PHP Version 7
+ *
+ * @category Tools
+ * @package  Loris
+ * @author   Zaliqa Rosli <zaliqa.rosli@mcin.ca>
+ * @licence  LORIS License
+ * @link     https://www.github.com/aces/Loris
+ */
+require_once __DIR__ . '/../generic_includes.php';
+
+$confirm = false;
+if (isset($argv[1]) && $argv[1] === 'confirm') {
+    $confirm = true;
+}
+
+echo "\n\nQuerying data...\n";
+$db      = \NDB_Factory::singleton()->database();
+// Query CommentIDs with DECS 'Complete', and
+// userID for latest changes made to the instrument data
+$history = $db->pselect(
```

Comment:
@ridz1208 if you could take a look at this query, i would appreciate it!

Reply:
@zaliqarosli i'm not sure I understand the need for the temporary table ? what is the role of `h2` here?

Reply:
the h2 gives all the max changeDate values for each commentID, and then i use that max date to query the row in the table that has that changeDate

Reply:
https://stackoverflow.com/a/12102216/10924787

Reply:
i'm essentially trying to make sure that the userID i'm querying is from the last data entry done on the instrument